### PR TITLE
Add New Play: Navbar

### LINKED ISSUE #307: [Add a Play]: Navbar using Tailwind
Thank You! 

## What's the Play about?
This is a simple responsive Navbar that is made with the help of Tailwind CSS.

## What ReactJS concept will be used to create this play?
Just simple React Concepts, JSX, and useState.

## Are you willing to take it up for implementation?
Yes

### PR DESCRIPTION
**Pull request by Shivam-Katare**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

This is a simple responsive navbar that I made with the help of **Tailwind**

Fixes #307 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings

### CODE REVIEW

**Comment by koustov** [comment]
@Shivam-Katare associate an issue first.

**Comment by Shivam-Katare** [comment]
> @Shivam-Katare associate an issue first.

Issue associated. #307

**Comment by koustov** [comment]
@Shivam-Katare, this looks cool. An early suggestion,  can you create a set of simple textual content for each of the links. It will give the viewer the impression of navigation as well.

**Comment by Sachin-chaurasiya** [comment]
> @Shivam-Katare, this looks cool. An early suggestion, can you create a set of simple textual content for each of the links. It will give the viewer the impression of navigation as well.

+1

File: src/plays/navbar/NavBar.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          {/* Your Code Starts Here */}
+          <div className='flex'>
+
+            <div className={`${open ? 'w-72' : 'w-20'} 
+            duration-500 h-screen p-5 pt-8 bg-slate-900 relative`}>
+
+              <img src={require('../navbar/assest/img.png')}
+                alt=''
+                className={`absolute cursor-pointer rounded-full -right-3 top-9 w-7 border-2 bg-cyan-500 ${!open && "rotate-180"}`}
+                onClick={() => setOpen(!open)} />
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/NavBar.jsx:31]: Can we move this logic to callback?

```js
const handleOnClick = ()=>setOpen((prevState)=>!prevState)
```

```suggestion
                onClick={handleOnClick} />
```

File: src/plays/navbar/NavBar.jsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+              <div className='flex gap-x-4 items-center'>
+                <img src={require('../navbar/assest/logo.png')}
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/NavBar.jsx:34]: Can we import the image at the top?

```js
import LogoImage from '../navbar/assest/logo.png'
```
```suggestion
                <img src={LogoImage}
```

File: src/plays/navbar/NavBar.jsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                  alt=''
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/NavBar.jsx:35]: Can we add some value to the alt attribute?

```suggestion
                  alt='play-logo'
```

File: src/plays/navbar/Navbar-image.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import titleOne from "./assest/Chart.png";
+import titleTwo from "./assest/Chat.png";
+import titleThree from "./assest/User.png";
+import titleFour from "./assest/Setting.png";
+import titleFive from "./assest/Search.png";
+
+const Menus = [
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/Navbar-image.jsx:7]: Can we change this variable name to `MenuItems`?
```suggestion
const MenuItems = [
```

File: src/plays/navbar/navBar.css [hunk 5]
```diff
@@ -0,0 +1 @@
+/* enter stlyes here */
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navBar.css:1]: Can we remove this file as it's an empty file?
we can add it later if we want to add some styles.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Left few comments.

**Comment by Shivam-Katare** [comment]
> @Shivam-Katare, this looks cool. An early suggestion, can you create a set of simple textual content for each of the links. It will give the viewer the impression of navigation as well.

Ok, I do this.

**Comment by Shivam-Katare** [comment]
> Left few comments.

Thanks @Sachin-chaurasiya for pointing out these mistakes, I will correct these mistakes.

**Comment by atapas** [comment]
> @Shivam-Katare, this looks cool. An early suggestion, can you create a set of simple textual content for each of the links. It will give the viewer the impression of navigation as well.

+1

**Comment by Shivam-Katare** [comment]
> @Shivam-Katare, this looks cool. An early suggestion, can you create a set of simple textual content for each of the links. It will give the viewer the impression of navigation as well.

Hey, @koustov I have added links for each NavItems. So, is it right or should I do something different? I also do some of the changes in the code, you can check that out too.

**Comment by Sachin-chaurasiya** [comment]
Hey @Shivam-Katare , Please rebase your branch with main.

File: src/plays/navbar/navListItems.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
+import React, { useState } from 'react'
+import Collasp from '../navbar/assest/img.png'
+import LogoImage from '../navbar/assest/logo.png'
+import img1 from "./assest/Chart.png"
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navListItems.js:4]: Can you give a meaningful name to these images?
```suggestion
import ChartImg from "./assest/Chart.png"
```

File: src/plays/navbar/navListItems.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+import img2 from "./assest/Chat.png"
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navListItems.js:5]:
```suggestion
import ChatImg from "./assest/Chat.png"
```

File: src/plays/navbar/navListItems.js [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+import img3 from "./assest/User.png"
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navListItems.js:6]:
```suggestion
import UserImg from "./assest/User.png"
```

File: src/plays/navbar/navListItems.js [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+import img4 from "./assest/Setting.png"
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navListItems.js:7]:
```suggestion
import SettingImg from "./assest/Setting.png"
```

File: src/plays/navbar/navListItems.js [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+import img5 from "./assest/Search.png"
```
Comment (Sachin-chaurasiya) [inline src/plays/navbar/navListItems.js:8]:
```suggestion
import SearchImg from "./assest/Search.png"
```

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
Requested a few changes.

**Comment by Shivam-Katare** [comment]
> Requested a few changes.

Completed those changes

File: src/meta/play-meta.js [hunk 11]
```diff
@@ -24,6 +24,8 @@ import {
   DynamicRoutes,
   SimpleLiveChat,
   FlipCardEffect,
+  NavBar,
+
```
Comment (atapas) [inline src/meta/play-meta.js:28]: @Shivam-Katare There is a major fix went in main on this file.. please take the latest.

**Review by atapas — CHANGES_REQUESTED** [review]

File: src/meta/play-meta.js (hunk 11 above)
Comment (Shivam-Katare) [reply src/meta/play-meta.js]: Took the latest in this file.

**Comment by atapas** [comment]
@Shivam-Katare Please pull from main for build to success

**Comment by Shivam-Katare** [comment]
> @Shivam-Katare Please pull from main for build to success

@atapas  Pulled from main.

**Comment by atapas** [comment]
Hi @Shivam-Katare 

We have some massive changes in the data store and we are changing how the play gets created for a better usability. As this play is interim, we need to make sure it migrates correctly. Could you please ping me on Discord so that I make sure everything is placed?

https://discord.gg/YbmXwU5f

**Comment by Shivam-Katare** [comment]
> Hi @Shivam-Katare
> 
> We have some massive changes in the data store and we are changing how the play gets created for a better usability. As this play is interim, we need to make sure it migrates correctly. Could you please ping me on Discord so that I make sure everything is placed?
> 
> https://discord.gg/YbmXwU5f

@atapas I have pulled from main and resolve merge conflict too, but don't understand that why isn't it deploying? Is there anything that I need to do in this?

**Comment by Shivam-Katare** [comment]
> @Shivam-Katare
> 
> This is the error:
> 
> ## Failed to compile.
> 17:53:44.143 |   17:53:44.143 | Attempted import error: 'getPlayById' is not exported from 'meta/play-meta-util' (imported as 'getPlayById'). 17:53:44.143 |   17:53:44.143 |   17:53:44.217 | error Command failed with exit code 1. 17:53:44.217 | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command. 17:53:44.225 | Error: Command "yarn run build" exited with 1

Okay, I will try to solve this issue.

**Review by koustov — APPROVED** [review]
lgtm

**Comment by Shivam-Katare** [comment]
> 

Thanks @koustov

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
Hey @Sachin-chaurasiya 

Could you please check this PR? It is requested review from you. Myself and Koustov have approved it already, once you do, I can merge it. Thanks!

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**